### PR TITLE
locations: adapt detail view with new fields

### PR DESCRIPTION
**Pull request by zannkukai**
* Adapts location resource detail view to display new fields defined into the json schema.
* Adapts "request item" select box to display only valid pickups location if needed
* Refactors 'generateTooltipMessage' function to be modular (working for other message type than 'deleteMessage')

Works with : https://github.com/rero/rero-ils/pull/708

## How to test?

- Edit a location to check edition of new fields "contact_email", "allow_request", "send_notification" and "restrict_pickup_to". Set some restrictions on "pickup_location" (choose one or two pickup locations). 
- Add an item for this location.
- Check the "request an item for a patron" for this item. The pikup location select box should only display the pickups you previously selected.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/circulation/items.service.ts [hunk 1]
```diff
@@ -143,4 +143,37 @@ export class ItemsService {
       })
       );
   }
+
+  /**
+   * Get the available pickup locations for an item.
+   * @param itemPid: the item pid to be requested
+   */
+  getPickupLocations(itemPid) {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/items.service.ts:151]: missing return type of the method + @return in doc block

File: projects/admin/src/app/circulation/items.service.ts [hunk 2]
```diff
@@ -143,4 +143,37 @@ export class ItemsService {
...
+    );
+  }
+
+  /**
+   * Check if an item can be requested.
+   * @param itemPid: the item pid to check
+   * @param libraryPid: the library_pid to check
+   * @param patronBarcode: the patron barcode to check
+   */
+  canRequest(itemPid: string, libraryPid?: string, patronBarcode?: string) {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/items.service.ts:169]: same here, missing return type.

File: projects/admin/src/app/circulation/items.ts [hunk 3]
```diff
@@ -176,3 +177,12 @@ export class Item {
     return (this.pending_loans && this.pending_loans.length > 0);
   }
 }
+
+export interface ItemRecordPermission extends RecordPermission {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/circulation/items.ts:181]: add doc block for explaining the interface.

File: projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts [hunk 4]
```diff
@@ -223,30 +218,20 @@ export class ItemRequestComponent implements OnInit {
...
-  /**
-   * Get pickups by organisation pid
-   * @param organisationPid - string
-   * @return observable
+
+  /** Get pickup location available for the item
+   *  @return observable
    */
-  private getPickupsByOrganisation(organisationPid: string) {
+  private getPickupLocations() {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts:225]: Add return type

File: projects/admin/src/app/service/record-permission.service.ts [hunk 5]
```diff
@@ -54,55 +54,67 @@ export class RecordPermissionService {
...
     }
 
-    messages.unshift(
-      messages.length === 1 ?
-        this._translateService.instant('You cannot delete the record for the following reason:') :
-        this._translateService.instant('You cannot delete the record for the following reasons:')
-    );
-
+    if (messages.length > 0 && messageType) {
+      switch (messageType) {
```
Comment (sebastiendeleze) [inline projects/admin/src/app/service/record-permission.service.ts:90]: What happens if messageType is not one of "delete" or "request"? I think a default behavior have to be added, or maybe throw an error if messageType is not one of them.

File: projects/admin/src/app/circulation/items.service.ts (hunk 1 above)
Comment (zannkukai) [reply projects/admin/src/app/circulation/items.service.ts:151]: done

File: projects/admin/src/app/circulation/items.service.ts (hunk 2 above)
Comment (zannkukai) [reply projects/admin/src/app/circulation/items.service.ts:169]: done

File: projects/admin/src/app/circulation/items.ts (hunk 3 above)
Comment (zannkukai) [reply projects/admin/src/app/circulation/items.ts:181]: as this interface is no more used, I delete it

File: projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts (hunk 4 above)
Comment (zannkukai) [reply projects/admin/src/app/record/detail-view/document-detail-view/item-request/item-request.component.ts:225]: done

File: projects/admin/src/app/service/record-permission.service.ts (hunk 5 above)
Comment (zannkukai) [reply projects/admin/src/app/service/record-permission.service.ts:90]: done

**Pull request closed without merging**